spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/converter/AbstractAzureMessageConverter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Abstract class handles common conversion logic between <T> and {@link Message}
+ *
+ * @author Warren Zhu
+ */
+public abstract class AbstractAzureMessageConverter<T> implements AzureMessageConverter<T> {
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public T fromMessage(Message<?> message, Class<T> targetClass) {
+        T azureMessage = internalFromMessage(message, targetClass);
+
+        setCustomHeaders(message, azureMessage);
+
+        return azureMessage;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <U> Message<U> toMessage(T azureMessage, MessageHeaders headers, Class<U> targetPayloadClass) {
+        return (Message<U>) internalToMessage(azureMessage, headers, targetPayloadClass);
+    }
+
+    protected abstract byte[] getPayload(T azureMessage);
+
+    protected abstract T fromString(String payload);
```

Comment:
how do we handle payload that's massive? we should treat the payload in the way of stream regardless it's text message or byte message.

e.g. MarshallingMessageConverter

Reply:
1. If payload is too large, then backed service should throw exception. Normally, service has its constraints of message size. 
2. `Stream` also has text-based and byte-based, we still need to distinguish.

Reply:
my ask is: we should we look into org.springframework.jms.support.converter. See if we could do better. Message conversion will be widely used in our repo.